packages/core/src/editor/NodeHelpers.ts

```diff
@@ -92,27 +91,29 @@ export function NodeHelpers(state: EditorState, id: NodeId) {
             return result;
           }
 
-          // Include linkedNodes if any
-          const linkedNodes = nodeHelpers(id).linkedNodes();
+          if (includeOnly !== "childNodes") {
+            // Include linkedNodes if any
+            const linkedNodes = nodeHelpers(id).linkedNodes();
 
-          linkedNodes.forEach((nodeId) => {
-            result.push(nodeId);
-            result = appendChildNode(nodeId, result, depth + 1);
-          });
-
-          const childNodes = node.data.nodes;
-
-          if (!childNodes) {
-            return result;
-          }
-
-          // Include child Nodes if any
-          if (childNodes) {
-            childNodes.forEach((nodeId) => {
+            linkedNodes.forEach((nodeId) => {
               result.push(nodeId);
               result = appendChildNode(nodeId, result, depth + 1);
             });
           }
+
+          if (includeOnly !== "linkedNodes") {
+            const childNodes = node.data.nodes;
+
+            // Include child Nodes if any
+            if (childNodes) {
+              childNodes.forEach((nodeId) => {
+                result.push(nodeId);
+                result = appendChildNode(nodeId, result, depth + 1);
+              });
+            }
+          }
+
+          return result;
```

Comment:
can we rename this result to 'descendantNodeIds' ?

Reply:
What about `descendants`? I think TS already makes it clear that it's `NodeId[]`

Reply:
or that, sure 👍